### DISCUSSION TITLE
Create Draft Listing Image Upload

### DISCUSSION THREAD

**Discussion by Esatyilmaz0** [etsy/open-api#919] [Ideas]
Hello, we're making an app and we can't upload an image while creating a draft list, because etsy wants an image id and it doesn't exist? What solution do you have for this? this parameter was not a required parameter in v2

**Comment by PMStanley** [comment]
I've not personally used the endpoint but isn't the answer to upload the image using https://developers.etsy.com/documentation/reference#operation/uploadListingImage at which point you have an ID you can use in the Listing creation?

**Comment by turboturtleD** [comment]
I recall that the listing draft needs to be created before you can upload images.

You do not need an image id, Etsy will automatically assign that.  If you need to reference that image later, you would save the response from the server regarding what image id is assigned upon successful completion of your image upload.

There are some quirks to using fetch regarding headers for uploading images - so if you are indeed using fetch, we discussed info on that many many months ago and could probably be dug up pretty easily with a search query.  That is not relevant specifically to image ids though, just uploading images using fetch.

**Comment by adg-0** [comment]
I just did this yesterday, and yes you need to create a draft first, then use the listing_id to upload an image. I also had the issue with the image_id which made no sense. IMO it's the API responses that are not up to date because it will say that if the serve doesn't like the way you're uploading the image.

Let's say I have created a read stream for my img :
```javascript
img = fs.createReadStream("/path/img.png")
``` 
I then built my object to upload like this :
```javascript
const imgForm = {
	image: {
		value: img,
		options: {
			filename: path.basename(img.path),
			contentType: null
		}
	}
}
```
Some properties are missing yes because some of them are getting "bad request" responses from their server.
Finally, for the actual request :
```javascript
	static uploadListingImage(api_key, token, shopId, listingID, imgForm) {
		var options = {
			method: 'POST',
			url: `https://openapi.etsy.com/v3/application/shops/${shopId}/listings/${listingID}/images`,
			headers: {
				'Content-Type': 'multipart/form-data',
				'x-api-key': api_key.key_string,
				'Authorization': `Bearer ${token.access_token}`	
			},
			formData: imgForm
		}
		return new Promise((resolve, reject) => {
			request(options, function (error, response) {
				if (error) reject(error)
				resolve(JSON.parse(response.body))
			})
		})
		.catch(error => this.handleError(error))
	}
```

There, I usually use "fetch" with Promises but fetch wouldn't work with this upload for some reason, only the package "request".
No idea if it's the optimal way to do it, but it's the only one I found to work

**Comment by turboturtleD** [comment]
As noted above, fetch does (can) work, but you need to omit a header.  I think the Content-type header needs to be omitted.  This is a fetch bug we believe.
